handler_api_endpoint.go

```diff
@@ -1112,6 +1112,24 @@ type PartitionConfig struct {
 	ByKind bool     `json:"by_kind"`
 }
 
+type RiverJobMinimal struct {
```

Comment:
Not sure if you have another preferred pattern for this, but with the prospect for larger amounts of data being stored in metadata it doesn't seem like a great idea to fetch or render the entire thing into the API response. This was just a quick way to get unblocked and put the problem forward.

Actually the query itself is untouched here since we're just leveraging `JobList`, so it's fetching the payloads whether or not they're used. But still, the frontend doesn't need to keep these for a potentially long list of jobs being displayed, and then re-fetching it every few seconds.

Reply:
It is indeed the idiomatic way to segregate "exported" data to the dedicated struct.

Reply:
We use a similar approach at work, even using the "minimal" moniker, but what do you think about embedding the minimal job on the main job like?

``` go
type RiverJob struct {
    RiverJobMinimal
}

The JSON fields end up at the same top level, but that way we don't need to maintain two completely duplicate lists of values, and it's obvious what's different in the minimal version versus non-minimal version.

Reply:
I couldn't remember whether it came out unnested or not in the JSON, so great news, I've made that switch.